documentation/modules/ROOT/pages/connectors/vitess.adoc

```diff
@@ -31,15 +35,44 @@ The connector gives you the flexibility to choose to subscribe to the `MASTER` n
+The underlying MySQL implementation in Vitess purges binary logs based on some configurable period of time. 
+Because the contents of the binlog might not be complete, the connector requires another mechanism to ensure that it captures the complete content of a particular database.
+Thus, when the connector connects to a database for the first time, it performs a consistent snapshot of the database. 
+After the connector completes the snapshot, it continues streaming changes from the exact point at which the snapshot was made. 
+In this way, the connector starts with a consistent view of all of the data, and does not omit any changes that were made while the snapshot was being taken.
+
+The connector is tolerant of failures. 
+As the connector reads changes and produces events, it records the VGTID position for each event. 
+If the connector stops for any reason (including communication failures, network problems, or crashes), after the connectors restarts, it continues reading the WAL from the point at which it stopped. 
+This behavior does not apply to snapshots. If the connector stops during a snapshot, after a restart, the connector does not continue performing snapshots where it last left off.
+We'll talk later about how the connector behaves xref:{link-vitess-connector}#vitess-when-things-go-wrong[when things go wrong].
 
 // Type: assembly
 // ModuleID: how-debezium-vitess-connectors-work
 // Title: How {prodname} Vitess connectors work
 [[how-the-vitess-connector-works]]
 == How the connector works
 
-To optimally configure and run a {prodname} Vitess connector, it is helpful to understand how the connector streams change events, determines Kafka topic names, and uses metadata.
+To optimally configure and run a {prodname} Vitess connector, it is helpful to understand how the connector performs snapshots, streams change events, determines Kafka topic names, and uses metadata.
+
+// Type: concept
+// ModuleID: how-debezium-vitess-connectors-perform-snapshot
+// Title: How {prodname} Vitess connectors perform a snapshot
+[[vitess-snapshot]]
+=== Snapshot
+
+Typically, a MySQL server is not configured to retain the complete history of the database in the binary logs.
+As a result, the connector is unable to read the entire history of the database from the binary logs. 
+For this reason, the first time that the connector starts, it performs an initial consistent snapshot of the database. 
+You can change this behavior by setting the xref:{link-vitess-connector}#vitess-property-snapshot-mode[`snapshot.mode` connector configuration property] to a value other than `initial`.
+This snapshot feature is built on link:https://vitess.io/docs/design-docs/vreplication/vstream/vscopy/[VStream Copy] introduced in version 7.0.
+
+[NOTE]
+====
+In Vitess 15 or earlier, it is not possible to resume `VStream Copy.`
+Starting in Vitess 16, it will support resuming this initial copy phase if it's interrupted for any reason.
```

Comment:
```suggestion
Beginning with this release of the connector, in environments that run Vitess 16 or later, if a snapshot is interrupted for any reason, after the connection is restored, the snapshot automatically resumes processing data from the point at which it stopped. In environments that run Vitess 15 or earlier, automatic retry is not available. 
```
@yoheimuta Let me know if this latest edit correctly captures the behavior. Per the new description, I assume that the snapshot resumes at the point at which it stopped, yes? That would seem to be the sensible thing. :)

Reply:
@roldanbob I'm concerned that users may misunderstand that the connector can resume the snapshot just if they update Vitess to 16.
The correct timeline for implementing this feature is as follows

1. Vitess team releases v16.
2. Someone (I or maintainer) updates a connector to build on Vitess 16 feature and cut the release, for example, Debezium v3.
3. Users begins using Vitess v16.
4. Users begins using Debezium v3.

To avoid confusion, it may be best to remove this note. 🙇‍♂️

Reply:
@roldanbob Thank you for your suggestion!
I added [5da7d4a](https://github.com/debezium/debezium/pull/4158/commits/5da7d4ac2746f8a65848ee0fe32a59bcbbbd19ea).